presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkInjectorFactory.java

```diff
@@ -33,18 +33,23 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
 
 import static com.facebook.presto.server.PrestoSystemRequirements.verifySystemTimeIsReasonable;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
 import static java.util.Objects.requireNonNull;
 
 public class PrestoSparkInjectorFactory
 {
-    private final Map<String, String> properties;
+    private final Map<String, String> configProperties;
+    private final Map<String, Map<String, String>> catalogProperties;
```

Comment:
Any reason why we don't use `Map<ConnectorId, Map<String, String>>` (see https://github.com/prestodb/presto/blob/20aa3228be1317c74659be6a57003b79e738ae55/presto-main/src/main/java/com/facebook/presto/SessionRepresentation.java#L58 ) ?

Reply:
The `StaticCatalogStore` takes catalog name as a string: https://github.com/prestodb/presto/pull/14152/files/4d8e7e658b13ce71c89f2fab515f5f461e640bb8#diff-665defb947f4d22b0cf16cb75130fcddR102 because further the ConnectorManager also takes the catalog name as a string: https://github.com/prestodb/presto/pull/14152/files/4d8e7e658b13ce71c89f2fab515f5f461e640bb8#diff-665defb947f4d22b0cf16cb75130fcddR124. The `ConnectorManager#createConnection` returns the `ConnectorId`.